packages/compass-e2e-tests/tests/collection-documents-tab.test.ts

```diff
@@ -221,43 +221,45 @@ describe('Collection documents tab', function () {
   for (const maxTimeMSMode of ['ui', 'preference'] as const) {
     it(`supports maxTimeMS (set via ${maxTimeMSMode})`, async function () {
       let maxTimeMSBefore;
-      if (maxTimeMSMode === 'preference') {
-        maxTimeMSBefore = await browser.getFeature('maxTimeMS');
+      try {
+        if (maxTimeMSMode === 'preference') {
+          maxTimeMSBefore = await browser.getFeature('maxTimeMS');
+
+          await browser.openSettingsModal();
+          const settingsModal = await browser.$(Selectors.SettingsModal);
+          await settingsModal.waitForDisplayed();
+          await browser.clickVisible(Selectors.GeneralSettingsButton);
+
+          await browser.setValueVisible(
+            Selectors.SettingsInputElement('maxTimeMS'),
+            '1'
+          );
+          await browser.clickVisible(Selectors.SaveSettingsButton);
+        }
 
-        await browser.openSettingsModal();
-        const settingsModal = await browser.$(Selectors.SettingsModal);
-        await settingsModal.waitForDisplayed();
-        await browser.clickVisible(Selectors.GeneralSettingsButton);
+        // execute a query that will take a long time, but set a maxTimeMS shorter than that
+        await browser.runFindOperation(
+          'Documents',
+          '{ $where: function() { return sleep(10000) || true; } }',
+          {
+            ...(maxTimeMSMode === 'ui' ? { maxTimeMS: '1' } : {}),
+            waitForResult: false,
+          }
+        );
 
-        await browser.setValueVisible(
-          Selectors.SettingsInputElement('maxTimeMS'),
-          '1'
+        const documentListErrorElement = await browser.$(
+          Selectors.DocumentListError
         );
-        await browser.clickVisible(Selectors.SaveSettingsButton);
-      }
+        await documentListErrorElement.waitForDisplayed();
 
-      // execute a query that will take a long time, but set a maxTimeMS shorter than that
-      await browser.runFindOperation(
-        'Documents',
-        '{ $where: function() { return sleep(10000) || true; } }',
-        {
-          ...(maxTimeMSMode === 'ui' ? { maxTimeMS: '1' } : {}),
-          waitForResult: false,
+        const errorText = await documentListErrorElement.getText();
+        expect(errorText).to.include(
+          'Operation exceeded time limit. Please try increasing the maxTimeMS for the query in the expanded filter options.'
+        );
+      } finally {
+        if (maxTimeMSMode === 'preference') {
+          await browser.setFeature('maxTimeMS', maxTimeMSBefore);
```

Comment:
I was gonna say that naybe also a good idea to just set this in a beforeEach so each test starts off with a known value and then you don't have to reason about whether a test failed and whether the cleanup ran. But I realise that might be slow. So this is probably fine.

The next test run should start with fresh default settings, right?

Reply:
Yeah, good point! I think it would be okay to move this in `beforeEach`: `setFeature` just calls electron ipc in render to tell preference model to update, so no UI interactions involved and should be snappy